src/components/connect.js

```diff
@@ -73,8 +74,23 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
     }
 
     class Connect extends Component {
-      shouldComponentUpdate() {
-        return !pure || this.haveOwnPropsChanged || this.hasStoreStateChanged
+
+      shouldComponentUpdate(nextProps,nextState) {
+
```

Comment:
Kind of hacky but if componentWillReceiveProps receive same props as prevProps, then we don't want to redo a shallowEqual here. I could have put the props in component state but not sure it would be better